rfcs/no_repeat_expected.md

```diff
@@ -0,0 +1,57 @@
+# RFC 119: Allow wptrunner to skip tests that run as expected
+
+## Summary
+
+Add a `--no-repeat-expected` flag to wptrunner to skip tests in subsequent
+repeat iterations once they run as expected.
+
+## Details
+
+### Rationale
+
+This flag is used with `--repeat` to implement a retry mechanism.
```

Comment:
It seems like this flag actually has two effects:
* Retries tests that got an unexpected result up to N times.
* Changes the result handling to prefer the expected result in case of multiple results (i.e for an expected PASS, FAIL, PASS will be treated just like PASS).

So rather than viewing this as a modofoer to `--repeat`, I suggest we just make a new flag like `--retry-unexpected=N` and document that it has the behaviour of taking any expected status as the final result.